### DISCUSSION TITLE
New `whereTrue()` and `whereFalse()` query methods

### DISCUSSION THREAD

**Discussion by troccoli** [laravel/framework#46071] [Polls]
Hello everyone,

We've got quite a few tables with some booleans columns, so we often use `where('is_paid', true)` and similar in our queries. But I think something like `whereTrue('is_pad')` is more readable.

This poll is to find out whether we do think it would be useful. If so, then I will create a PR and we then discuss the name of the methods, if we want some alias, etc.

**Comment by kingIZZZY** [comment]
This could also have implication on the generated SQL?
`where('is_paid', true)` = `where is_paid = 1`
`whereTrue('is_paid')` = `where is_paid = true`

Or even affect the resulting SQL functionally (albeit only slightly) ?
`whereTrue('is_paid')` = `where is_paid` 👉 evaluates the expression itself if truthy or falsy

**Reply by shaedrich** [reply]
This could be handled accordingly by the sql engine's implementation of Laravel if necessary.

**Comment by TheBlckbird** [comment]
I think this could be problematic because you could write something like this: `User::where("name","TheBlckbird")` like this: `User::whereName("TheBlckbird")`.  
This means that `whereTrue("is_paid")` would be `where("true", "is_paid")`.

**Edit:**
I just realized that there are already modifiers that do this: `whereColumn('destination_id', 'destinations.id')` ([laravel.com/docs/10.x/eloquent#subquery-selects](https://laravel.com/docs/10.x/eloquent#subquery-selects)), so it probably wouldn't be a problem.

**Comment by KIOlpHL** [comment]
cxvxg

**Comment by ahinkle** [comment]
So it's been submitted a few times. 
https://github.com/laravel/framework/pull/43088 https://github.com/laravel/framework/pull/46307

Would love to see this though.
